components/sup/src/manager/service/mod.rs

```diff
@@ -366,7 +366,17 @@ impl Service {
     fn all_binds_satisfied(&self, census_ring: &CensusRing) -> bool {
         let mut ret = true;
         for ref bind in self.binds.iter() {
-            if census_ring.census_group_for(&bind.service_group).is_none() {
+            if let Some(group) = census_ring.census_group_for(&bind.service_group) {
+                if group.members().iter().all(|m| !m.alive()) {
```

Comment:
If we're splitting hairs, and we aren't but I'm going to 😅 , you can use `.any()` to abort the iterator on the first match instead of iterating to the end